ysoserial/Generators/ActivitySurrogateDisableTypeCheck.cs

```diff
@@ -68,7 +68,8 @@ public override object Generate(string formatter, InputArgs inputArgs)
                 xaml_payload = XMLMinifier.Minify(xaml_payload, null, null);
             }
 
-            TextFormattingRunPropertiesMarshal payload = new TextFormattingRunPropertiesMarshal(xaml_payload);
+            // TextFormattingRunPropertiesMarshal payload = new TextFormattingRunPropertiesMarshal(xaml_payload);
+            object payload = TypeConfuseDelegateGenerator.GetXamlGadget(xaml_payload);
```

Comment:
Can you move `object payload = TypeConfuseDelegateGenerator.GetXamlGadget(xaml_payload);` to the top of the method and then add `payload = new TextFormattingRunPropertiesMarshal(xaml_payload);` in the `if (inputArgs.Minify)` block?

This way, if the user wants a short payload, it will get one using TextFormatting gadget, and if he wants a more reliable one regardless of the size, he can choose the XamlReader one

Reply:
Just modified it